packages/ethers/src/router/get-message-and-data-by-tx-hash.ts

```diff
@@ -43,7 +43,10 @@ export async function getMessagesByTxHash(
 
   return receipt.logs.flatMap((log) => {
     if (log.address.toLowerCase() === routerContractAddress.toLowerCase()) {
-      return [decodeLogMessageSendRequested(log)];
+      const equitoMessageWithData = decodeLogMessageSendRequested(log);
+      if (equitoMessageWithData) {
+        return [equitoMessageWithData];
```

Comment:
```suggestion
        return equitoMessageWithData;
```

this is more readable and should work as well